src/Shell/Task/TemplateTask.php

```diff
@@ -422,46 +423,12 @@ public function getOptionParser()
     }
 
     /**
-     * Returns associations for controllers models.
-     *
-     * @param Table $model The model to build associations for.
-     * @return array associations
+     * Get filtered associations
+     * To be mocked...
+     * @param Table $model
      */
-    protected function _associations(Table $model)
+    protected function _filteredAssociations(Table $model)
     {
-        $keys = ['BelongsTo', 'HasOne', 'HasMany', 'BelongsToMany'];
-        $associations = [];
-
-        foreach ($keys as $type) {
-            foreach ($model->associations()->type($type) as $assoc) {
-                $target = $assoc->target();
-                $assocName = $assoc->name();
-                $alias = $target->alias();
-                $targetClass = get_class($target);
-                list(, $className) = namespaceSplit($targetClass);
-
-                $modelClass = get_class($model);
-                if ($modelClass !== 'Cake\ORM\Table' && $targetClass === $modelClass) {
-                    continue;
-                }
-
-                $className = preg_replace('/(.*)Table$/', '\1', $className);
-                if ($className === '') {
-                    $className = $alias;
-                }
-
-                $associations[$type][$assocName] = [
-                    'property' => $assoc->property(),
-                    'variable' => Inflector::variable($assocName),
-                    'primaryKey' => (array)$target->primaryKey(),
-                    'displayField' => $target->displayField(),
-                    'foreignKey' => $assoc->foreignKey(),
-                    'alias' => $alias,
-                    'controller' => $className,
-                    'fields' => $target->schema()->columns(),
-                ];
-            }
-        }
-        return $associations;
+        return AssociationFilter::filterAssociations($model);
```

Comment:
Why a static method? Couldn't this be an instance? Static methods have generally been painful in the past as they cannot be mocked if required.